File: Frontend/src/app/models/maker/tax-file.ts
import { SafeResourceUrl } from "@angular/platform-browser";
import { Transaction } from "./transaction";

export class TaxFile {
    Id?: String;
    fileName?: String;
    extension?: String;
    supportId?: String;
    file?: Blob;
    fileType?: string;         // type of file (image/pdf/word)
    pdfUrl?: string;           // blob URL for PDF download
    safePdfUrl?: SafeResourceUrl;
    tax_id?: number;
}

File: Backend/TaxManagementSystem/src/main/java/com/tms/Maker/entity/TaxFile.java
package com.tms.Maker.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class TaxFile {
    private String Id;
    private String fileName;
    private String extension;
    private String supportId;
    private Long tax_id;
}
